src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -114,20 +126,49 @@ static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> f
                 return !failedList.Any();
             }
 
+            static bool analyzeSymbols(ElfBinary binary, out List<SymbolEntry<ulong>> failedList)
+            {
+                failedList = new List<SymbolEntry<ulong>>();
+
+                var symbols = binary.ELF.Sections.FirstOrDefault(s => s.Type == SectionType.DynamicSymbolTable) as SymbolTable<ulong>;
+                foreach (SymbolEntry<ulong> symbol in symbols.Entries)
+                {
+                    if (symbol.Name != "__stack_chk_fail" && symbol.Name != "stack_chk_guard" && symbol.Name != "__intel_security_cookie")
+                    {
+                        failedList.Add(symbol);
+                    }
+                    else
+                    {
+                        return true;
```

Comment:
>return [](http://example.com/codeflow?start=24&length=6)

So, what's our guarantee at this return location that failedList is empty?

Reply:
I might be misunderstanding the purpose of failedList here. I was basing my usage off of `analyzeDwarf`, which appears to simply store any command line args that are not stack-protector related in `failedList`. So, I'm simply storing any symbols that are not stack check related in my failedList. This might not actually be helpful for anything down the line, I just wanted to make this check as similar as possible to the existing one. I would expect there will be items in failedList for all binaries.